src/fluent-bit.c

```diff
@@ -718,6 +733,164 @@ static struct flb_cf *service_configure(struct flb_cf *cf,
     return cf;
 }
 
+#ifdef FLB_HAVE_CHUNK_TRACE
+static struct flb_input_instance *find_input(flb_ctx_t *ctx, const char *name)
+{
+    struct mk_list *head;
+    struct flb_input_instance *in;
+
+
+    mk_list_foreach(head, &ctx->config->inputs) {
+        in = mk_list_entry(head, struct flb_input_instance, _head);
+        if (strcmp(name, in->name) == 0) {
+            return in;
+        }
+        if (in->alias) {
+            if (strcmp(name, in->alias) == 0) {
+                return in;
+            }
+        }
+    }
+    return NULL;
+}
+
+static int enable_trace_input(flb_ctx_t *ctx, const char *name, const char *prefix, const char *output_name, struct mk_list *props)
+{
+    struct flb_input_instance *in;
+
+
+    in = find_input(ctx, name);
+    if (in == NULL) {
+        return FLB_ERROR;
+    }
+
+    flb_chunk_trace_context_new(in, output_name, prefix, NULL, props);
+    return (in->chunk_trace_ctxt == NULL ? FLB_ERROR : FLB_OK);
+}
+
+static int disable_trace_input(flb_ctx_t *ctx, const char *name)
+{
+    struct flb_input_instance *in;
+    
+
+    in = find_input(ctx, name);
+    if (in == NULL) {
+        return FLB_ERROR;
+    }
+
+    if (in->chunk_trace_ctxt != NULL) {
+        flb_chunk_trace_context_destroy(in);
+    }
+    return FLB_OK;
+}
+
+static int set_trace_property(struct mk_list *props, char *kv)
+{
+    int len;
+    int sep;
+    char *key;
+    char *value;
+
+    len = strlen(kv);
+    sep = mk_string_char_search(kv, '=', len);
+    if (sep == -1) {
+        return -1;
+    }
+
+    key = mk_string_copy_substr(kv, 0, sep);
+    value = kv + sep + 1;
+
+    if (!key) {
+        return -1;
+    }
+
+    flb_kv_item_create_len(props,
```

Comment:
I don't see a corresponding destroy for this, are we sure this is being cleaned up properly?